### ISSUE TITLE
"Assertion failed" as soon as it encounters an .SRT

### ISSUE THREAD

**Issue by tordenflesk** [ptr727/PlexCleaner#73]
Log:
https://pastebin.com/raw/7G42Pii7
Docker Stack:
```
version: '3.3'
services:
    plexcleaner:
        image: ptr727/plexcleaner
        container_name: plexcleaner
        environment:
          - PUID=1000
          - PGID=100
          - TZ=Europe/Oslo
        command: /PlexCleaner/PlexCleaner --settingsfile /config/PlexCleaner.json process --mediafiles /DATA00/media
        volumes:
          - /srv/dev-disk-by-label-DATA01/Docker_Config/PLEXCLEANER:/config
          - /srv/dev-disk-by-label-DATA00:/DATA00/media:rw
```
PlexCleaner.json:
```
{
  "SchemaVersion": 1,
  "ToolsOptions": {
    "UseSystem": false,
    "RootPath": ".\\Tools\\",
    "RootRelative": true,
    "AutoUpdate": true
  },
  "ConvertOptions": {
    "EnableH265Encoder": false,
    "VideoEncodeQuality": 20,
    "AudioEncodeCodec": "aac"
  },
  "ProcessOptions": {
    "DeleteEmptyFolders": true,
    "DeleteUnwantedExtensions": false,
    "KeepExtensions": ".partial~",
    "ReMux": false,
    "ReMuxExtensions": ".avi,.m2ts,.ts,.vob,.mp4,.m4v,.asf,.wmv",
    "DeInterlace": true,
    "ReEncode": false,
    "ReEncodeVideoFormats": "mpeg2video,mpeg4,msmpeg4v3,msmpeg4v2,vc1,h264",
    "ReEncodeVideoCodecs": "*,dx50,div3,mp42,*,*",
    "ReEncodeVideoProfiles": "*,*,*,*,*,Constrained Baseline@30",
    "ReEncodeAudioFormats": "flac,mp2,vorbis,wmapro,pcm_s16le",
    "SetUnknownLanguage": false,
    "DefaultLanguage": "eng",
    "RemoveUnwantedLanguageTracks": true,
    "KeepLanguages": "eng,nor,swe,dan",
    "RemoveDuplicateTracks": true,
    "PreferredAudioFormats": "truehd atmos,truehd,dts-hd master audio,dts-hd high resolution audio,dts,e-ac-3,ac-3",
    "RemoveTags": true,
    "UseSidecarFiles": true,
    "SidecarUpdateOnToolChange": false,
    "Verify": false,
    "FileIgnoreList": []
  },
  "MonitorOptions": {
    "MonitorWaitTime": 60,
    "FileRetryWaitTime": 5,
    "FileRetryCount": 2
  },
  "VerifyOptions": {
    "AutoRepair": true,
    "DeleteInvalidFiles": false,
    "RegisterInvalidFiles": true,
    "MinimumDuration": 300,
    "VerifyDuration": 0,
    "IdetDuration": 0,
    "MaximumBitrate": 100000000,
    "MinimumFileAge": 0
  }
}

```

**Comment by ptr727** [comment]
I've not tested with external subtitles.
Can you try to add `.srt` to `"KeepExtensions": ".partial~,.srt",`?

**Comment by tordenflesk** [comment]
Yes, I've added .srt and .tmp and haven't had a crash in a while.

**Comment by ptr727** [comment]
TODO: Add list of external subtitles to ignore / process:
E.g. https://support.plex.tv/articles/200471133-adding-local-subtitles-to-your-media/

**Comment by ptr727** [comment]
Fixed in version 2.3, that added more robust ignore and delete logic.
